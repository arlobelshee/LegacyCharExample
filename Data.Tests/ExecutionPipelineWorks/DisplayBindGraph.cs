using System;
using ApprovalTests;
using Data.PipelineSynchronous;
using NUnit.Framework;

namespace Data.Tests.ExecutionPipelineWorks
{
	[TestFixture]
	public class DisplayBindGraph
	{
		[Test]
		public void ShouldDisplayBindGraph()
		{
			var testSubject = new PipeSource<decimal, int>(_One);
			testSubject.AndThen(_Two).AndThen(new Collector<float>());
			testSubject.AndThen<string>(_Three);
			Approvals.Verify(testSubject);
		}

		private int _One(decimal arg)
		{
			throw new System.NotImplementedException();
		}

		private float _Two(int arg)
		{
			throw new System.NotImplementedException();
		}

		private void _Three(int arg, Action<string> generateOne)
		{
			throw new System.NotImplementedException();
		}
	}
}
