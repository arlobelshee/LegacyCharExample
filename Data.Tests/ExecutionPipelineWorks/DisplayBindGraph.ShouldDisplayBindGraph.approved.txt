--> Int32 DisplayBindGraph._One(Decimal arg)
 |--> Collector